libasherah.go

```diff
@@ -69,118 +81,31 @@ func SetupJson(configJson unsafe.Pointer) int32 {
 	config := AsherahConfig{}
 
 	configJsonStr, result := cobhan.BufferToString(configJson)
-	if result != 0 {
+	if result != ERR_NONE {
+		StdoutDebugOutput("Failed to deserialize configuration string")
 		return result
 	}
 
 	err := json.Unmarshal([]byte(configJsonStr), &config)
 	if err != nil {
-		StdoutDebugOutput("Failed to deserialize: " + err.Error())
-		return -100
+		StdoutDebugOutput("Failed to deserialize configuration JSON: " + err.Error())
+		return cobhan.ERR_JSON_DECODE_FAILED
```

Comment:
We should not propagate errors from dependencies. I would suggest defining it as an Asherah error code that other languages consumers of this library can rely on. That way we also avoid potential collusion of error codes (ERR_JSON_DECODE_FAILED is -5 in cobhan).